packages/preview-middleware-client/src/flp/init.ts

```diff
@@ -7,6 +7,9 @@ import IconPool from 'sap/ui/core/IconPool';
 import ResourceBundle from 'sap/base/i18n/ResourceBundle';
 import AppState from 'sap/ushell/services/AppState';
 import { getManifestAppdescr } from '../adp/api-handler';
+import VersionInfo from 'sap/ui/VersionInfo';
+import { getError } from '../cpe/error-utils';
+import initConnectors from './initConnectors';
```

Comment:
That line causes ui5 tooling served via npmjs (so without proxy) to no longer work because `LrepConnector.js` is not being served via UI5 tooling / npmjs

![image](https://github.com/user-attachments/assets/abdd92ae-8014-43ef-af8b-86a99b2403bc)

![image](https://github.com/user-attachments/assets/08ffd81e-e19f-4fee-838b-1237dbd21db3)

see also https://www.npmjs.com/package/@openui5/sap.ui.fl?activeTab=code

I don't yet understand why this is not an issue when serving the UI5 sources via proxy because
https://sapui5.hana.ondemand.com/1.124.0/resources/sap/ui/fl/LrepConnector.js also does not exist 🤷🏻 
(whereas https://sapui5.hana.ondemand.com/1.124.0/resources/sap/ui/fl/FakeLrepConnector.js does)

What would work is https://sapui5.hana.ondemand.com/1.124.0/resources/sap/ui/fl/initial/_internal/connectors/LrepConnector.js

I don't see any of those request when running via proxy

Reply:
Wouldn't a a trycatch around the execution of `initConnectors` "solve" the problem?

Reply:
Ok, seems like when served via npmjs the version is the app version from package.json❗

<img width="350" alt="image" src="https://github.com/user-attachments/assets/8a174c59-9a7d-4ab9-9fb2-7003fe215489">

loading that directly from CDN it contains the SAPUI5 Distribution version (which is ok as it comes directly from UI5)

<img width="451" alt="image" src="https://github.com/user-attachments/assets/a0ad4791-8b81-4e10-b96d-bb4dbebd9d4a">

So the `version` property we use is simply the wrong one. We should use the version of the sap.fe.core library

<img width="353" alt="image" src="https://github.com/user-attachments/assets/2f78c730-fd0d-4a6f-a18e-b1a8dc0c2c43">

<img width="453" alt="image" src="https://github.com/user-attachments/assets/c40ba0ff-0ab1-4cdd-b0af-1aeb8459966a">

That should fix it

Reply:
Wow, that is in interesting finding. It is actually a bigger problem then just the 2.0 enablement since it already causes a wrong behavior in other scenarios (that were just not as dramatic as now).

So, if I understood correctly, the correct solution would be to not look at the top level version but look at the libraries, find `sap.ui.core` and then take the version from there. That would work for both, a CDN ui5 version as well as something running locally with the modules from npmjs. @heimwege did I get it correctly?

@cfg74 should it get fix in this PR or is later ok? It will break the preview-middleware usage for runs with the `ui5-local.yaml` in Fiori tools projects.

Reply:
yep that's also my understanding 👍🏻

Reply:
There are quite some locations where the version is being used like this. Maybe having this in a central place would be a good idea

Reply:
@tobiasqueck breaking things is bad, why would it break the preview from downloaded sources?

Reply:
@cfg74 because the UI5 version handling is currently (and most likely has ever been) wrong. As of now it was assumed that the top level version from `sap-ui-version.json` represents the UI5 version but it represents the application version (from the application manifest `sap.app.applicationVersion.version`). Only when loading that file [directly from UI5](https://sapui5.hana.ondemand.com/resources/sap-ui-version.json) (e.g. via proxy and thus without the context of an actual application) the top level version equals the UI5 version.

In my case the application version is 1.0.0 and then tooling assumes this is my UI5 version and strange things happen 🙈

Reply:
But I'm working on fixing this and as son as @tobiasqueck agrees (because its contains changes in a lot of files and I'm not sure if all of this should be part of this PR) I can push my changes. Locally it seems to work pretty fine with the adjustments

Reply:
@heimwege @tobiasqueck clarified with Tobias, that _without_ the fix it will break local preview, hence we have to do the fix to avoid a regression. Can one of you take that please?

Reply:
I just pushed my changes. Hopefully that will fix it 🤞🏻
At least a test on my local machine was successfully with and without proxy

Reply:
I have tested it locally and it works like a charm!